### PR TITLE
Display subscription operation as the resource path in analytics graphs

### PR DESCRIPTION
**Pull request by msm1992**
Screenshot 2022-03-16 at 17 30 40

**Approach**
* A new field ctx was introduced to the  inboundMessageContext to hold ChannelHandlerContext
* API_ELECTED_RESOURCE is set as a property to ctx

Related Issue : https://github.com/wso2/product-apim/issues/12197

Note : The wildcard entries in the screenshot are from invocations before applying this fix.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessor.java [hunk 1]
```diff
@@ -87,6 +88,8 @@ public InboundProcessorResponseDTO handleRequest(int msgSize, String msgText,
                             // subscription operation name
                             String subscriptionOperation = GraphQLProcessorUtil.getOperationList(operation,
                                     inboundMessageContext.getGraphQLSchemaDTO().getTypeDefinitionRegistry());
+                            WebSocketUtils.setApiPropertyToChannel(inboundMessageContext.getCtx(),
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessor.java:91]: Dont we need to add the same fix to GraphQLResponseProcessor in handleResponse()? If someone use the same connection for two different subscriptions what would be the behavior?

**Comment by dushaniw** [comment]
@msm1992 in the screenshot the entry with 4 hits for the same API, were these added before the fix ?

**Comment by msm1992** [comment]
Yes, those 4 entries were there before this fix as well.

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessor.java (hunk 1 above)
Comment (msm1992) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessor.java:91]: fixed with https://github.com/wso2/carbon-apimgt/pull/11166/commits/fff020f1e84428a67f6601c736bbcbc1b2ca6cf9. Tested with multiple subscriptions in the same connections and shown below is show it looks in portal, correct resource path is mentioned in the table.
Screenshot 2022-03-18 at 06 35 00

**Review by dushaniw — APPROVED** [review]

**Review by Krishanx92 — APPROVED** [review]

**Pull request merged**